lib/web/JsonpChunkTemplatePlugin.js

```diff
@@ -30,6 +31,47 @@ class JsonpChunkTemplatePlugin {
 				const jsonpFunction = chunkTemplate.outputOptions.jsonpFunction;
 				const globalObject = chunkTemplate.outputOptions.globalObject;
 				const source = new ConcatSource();
+				let externals = [];
+
+				if (!chunk.entryModule) {
+					externals = []
+						.concat(
+							...chunk
+								.getModules()
+								.map(m =>
+									m.dependencies.filter(
+										d => d.module && d.module.externalType === "amd"
+									)
+								)
+						)
+						.map(d => d.module);
+
+					externals = externals.filter(function(value, index, self) {
+						return self.indexOf(value) === index && value;
```

Comment:
Would you mind moving the `value` to start of the conditional to potentially avoid an unnecessary  call to `indexOf`